pkg/costmodel/costmodel.go

```diff
@@ -708,7 +708,7 @@ func findDeletedPodInfo(cli prometheusClient.Client, missingContainers map[strin
 	if len(missingContainers) > 0 {
 		queryHistoricalPodLabels := fmt.Sprintf(`kube_pod_labels{}[%s]`, window)
 
-		podLabelsResult, _, err := prom.NewContext(cli).QuerySync(queryHistoricalPodLabels)
+		podLabelsResult, _, err := prom.NewNamedContext(cli, prom.ComputeCostDataContextName).QuerySync(queryHistoricalPodLabels)
```

Comment:
I'm not sure I understand the question

Reply:
`findDeletedPodInfo` is a private function using a `prometheusClient.Client` from somewhere else. What do you think of passing a named context to private functions from public functions, thus tying a context to a public API, instead of just re-creating the named context within the private function?

Reply:
I see, these APIs are ancient and need some love anyways, probably extends outside the scope of this PR.

Reply:
:+1: No reason to mess with it now, really. I think we'll discover if this is useful in time and then maybe rework this later if useful.